lbz/events/api.py

```diff
@@ -65,14 +84,23 @@ def get_all_failed_events(self) -> List[BaseEvent]:
         return self._failed_events
 
     def send(self) -> None:
-        try:
-            if self._pending_events:
-                entries = [self._create_eb_entry(event) for event in self._pending_events]
+        self._sent_events = []
+        self._failed_events = []
+
+        while self._pending_events:
+            events = self._pending_events[:MAX_EVENTS_TO_SEND_AT_ONCE]
+            try:
+                entries = [self._create_eb_entry(event) for event in events]
                 client.eventbridge.put_events(Entries=entries)
-            self._mark_sent()
-        except Exception as err:
-            self._mark_failed()
-            raise err
+                self._sent_events.extend(events)
+            except Exception as err:  # pylint: disable=broad-except
+                self._failed_events.extend(events)
+                logger.exception(err)
+
+            self._pending_events = self._pending_events[MAX_EVENTS_TO_SEND_AT_ONCE:]
+
+        if self._failed_events:
+            raise RuntimeError("Sending events has failed. Check logs for more details!")
```

Comment:
_FYI:_ I was considering creating a custom Exception for this situation but it looks that lbz is raising mostly HTTP-like errors or `RuntimeError`, so I decided to stay consistent 😉 

This way or another, each error will be logged individually by the 98. line.